apps/next-product-starter/src/pages/player.tsx

```diff
@@ -1,28 +1,79 @@
 import * as React from 'react'
 import Layout from '@skillrecordings/react/dist/layouts'
 import {Player, VideoProvider, HLSSource} from '@skillrecordings/player'
+import {first, noop} from 'lodash'
+
+type VideoResource = {
+  title: string
+  url: string
+  subtitlesUrl?: string
+  notesUrl?: string
+}
 
 const PlayerPage = () => {
-  const videoUrl = `https://d2c5owlt6rorc3.cloudfront.net/egghead-create-a-new-supabase-project-lEG2O_feW/hls/egghead-create-a-new-supabase-project-lEG2O_feW.m3u8`
-  const subtitlesUrl = `https://app.egghead.io/api/v1/lessons/javascript-create-a-new-supabase-project/subtitles`
-  const notesUrl = `https://gist.githubusercontent.com/joelhooks/d16c649f7d258289cbbc622d8867c608/raw/45f875698610140c28ec991c1aad1979dbf1a990/test.vtt`
+  const videos: VideoResource[] = [
+    {
+      title: 'Create a New Supabase Project',
+      url: `https://d2c5owlt6rorc3.cloudfront.net/egghead-create-a-new-supabase-project-lEG2O_feW/hls/egghead-create-a-new-supabase-project-lEG2O_feW.m3u8`,
+      subtitlesUrl: `https://app.egghead.io/api/v1/lessons/javascript-create-a-new-supabase-project/subtitles`,
+      notesUrl: `https://gist.githubusercontent.com/joelhooks/d16c649f7d258289cbbc622d8867c608/raw/45f875698610140c28ec991c1aad1979dbf1a990/test.vtt`,
+    },
+    {
+      title: 'Random Video',
+      url: 'https://bitdash-a.akamaihd.net/content/MI201109210084_1/m3u8s/f08e80da-bf1d-4e3d-8899-f0f6155f6efa.m3u8',
+    },
+  ]
+
+  const [currentVideo, setCurrentVideo] = React.useState<VideoResource>(
+    videos[0],
+  )
+
   return (
     <Layout>
       <VideoProvider>
         <Player className="font-sans">
-          <HLSSource src={videoUrl} />
-          <track
-            src={subtitlesUrl}
-            kind="subtitles"
-            srcLang="en"
-            label="English"
-            default
-          />
-          <track id="notes" src={notesUrl} kind="metadata" label="notes" />
+          <HLSSource src={currentVideo.url} />
+          {currentVideo.subtitlesUrl && (
+            <track
+              src={currentVideo.subtitlesUrl}
+              kind="subtitles"
+              srcLang="en"
+              label="English"
+              default
+            />
+          )}
+          {currentVideo.notesUrl && (
+            <track
+              id="notes"
+              src={currentVideo.notesUrl}
+              kind="metadata"
+              label="notes"
+            />
+          )}
         </Player>
       </VideoProvider>
+      <VideoResourceList>
+        {videos.map((videoResource) => {
```

Comment:
my first instinct was to pass in the `videos` as a prop to the `VideoResourceList` but this should primarily be a layout concern and the *shape of the data* isn't something that it needs to be specifically concerned with.

so instead the list is a wrapper and the end-user is responsible for actually styling the row based on the data that exists for the specific resource.

this could also mean that non-video resources might be present as well.